MovingBlocks/Terasology
# remove snapshot-or-not logic from gradle configuration

This removes the code from the build configurations that would decide whether or not to call a module build a snapshot based on the current git branch, because #4037.

It also includes #4032, which is bad for separation of concerns, but good for reducing the number of times we have to go through the cycle of review and maybe-rebuild-a-lot-of-modules.


### How to test

build some modules,  
make sure the version in the name of the jar file matches the one in its source module.txt, whether or not it has -SNAPSHOT.

maybe run the `generatePomFile` task and look at the `<version>` element in there too.

there's also some stuff about choosing which repository we publish to that should probably be tested by someone who knows how the repositories work.

- [ ] https://github.com/MovingBlocks/Terasology/blob/ab028b2d5a8557b351e1c9a10ff051640ca8d176/config/gradle/publish.gradle#L36-L39

### Outstanding before merging

- [x] `cacheChangingModulesFor` setting?
  - [x] in root: https://github.com/MovingBlocks/Terasology/blob/ab028b2d5a8557b351e1c9a10ff051640ca8d176/build.gradle#L179-L182
  - [x] in engine https://github.com/MovingBlocks/Terasology/blob/ab028b2d5a8557b351e1c9a10ff051640ca8d176/engine/build.gradle#L87-L89
  - [x] in modules: https://github.com/MovingBlocks/Terasology/blob/ab028b2d5a8557b351e1c9a10ff051640ca8d176/buildSrc/src/main/kotlin/terasology-module.gradle.kts#L88-L90
- [x] snapshot determination for engine and engine-test? https://github.com/MovingBlocks/Terasology/blob/012985ccb97b6872af9d5bfdb3c589845ab98605/engine/build.gradle#L241-L243
